vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -119,212 +117,137 @@ class OidcSiopWallet(
-            val params = AuthenticationRequestParameters.deserialize(jws.payload.decodeToString()).getOrElse {
-                return null
-                    .apply { Napier.w("parseRequestObjectJws: Deserialization failed", it) }
-            }
-            if (requestObjectJwsVerifier.invoke(jws, params))
-                AuthenticationRequestParametersFrom.JwsSigned(jws, params)
-            else null
-                .also { Napier.w("parseRequestObjectJws: Signature not verified for $jws") }
-        }
-    }
+    suspend fun parseAuthenticationRequestParameters(input: String) =
+        AuthenticationRequestParser.createWithDefaults(
+            remoteResourceRetriever = remoteResourceRetriever,
+            requestObjectJwsVerifier = requestObjectJwsVerifier,
+        ).parseAuthenticationRequestParameters(input)
 
     /**
      * Pass in the deserialized [AuthenticationRequestParameters], which were either encoded as query params,
      * or JSON serialized as a JWT Request Object.
      */
     suspend fun createAuthnResponse(
-        request: AuthenticationRequestParametersFrom
+        request: AuthenticationRequestParametersFrom,
     ): KmmResult<AuthenticationResponseResult> = catching {
-        val response = createAuthnResponseParams(request).getOrThrow()
-        request.parameters.responseType?.let {
-            if (!it.contains(ID_TOKEN) && !it.contains(VP_TOKEN))
-                throw OAuth2Exception(Errors.INVALID_REQUEST)
-                    .also { Napier.w("createAuthnResponse: Unknown response_type $it") }
-        } ?: throw OAuth2Exception(Errors.INVALID_REQUEST)
-            .also { Napier.w("createAuthnResponse: response_type null in ${request.parameters}") }
-
-        when (request.parameters.responseMode) {
-            DIRECT_POST -> authnResponseDirectPost(request, response)
-            DIRECT_POST_JWT -> authnResponseDirectPostJwt(request, response)
-            QUERY -> authnResponseQuery(request, response)
-            FRAGMENT, null -> authnResponseFragment(request, response)
-            is OTHER -> TODO()
-        }
+        AuthenticationResponseFactory(jwsService).createAuthenticationResponse(
+            request,
+            response = createAuthnResponseParams(request).getOrThrow(),
+        )
     }
 
-    private fun authnResponseDirectPost(
-        request: AuthenticationRequestParametersFrom,
-        response: AuthenticationResponse
-    ): AuthenticationResponseResult.Post {
-        val url = request.parameters.responseUrl
-            ?: request.parameters.redirectUrl
-            ?: throw OAuth2Exception(Errors.INVALID_REQUEST)
-        return AuthenticationResponseResult.Post(url, response.params.encodeToParameters())
+    /**
+     * Creates the authentication response from the RP's [params]
+     */
+    suspend fun createAuthnResponseParams(
+        params: AuthenticationRequestParametersFrom
+    ): KmmResult<AuthenticationResponse> = catching {
```

Comment:
can't we  `map` the result?

Reply:
fixed in commit e0d9d1f448615405bd526b2e676580e2b90ae884